device/thunder_ripple_sdk/src/client/device_operator.rs

```diff
@@ -37,13 +38,46 @@ pub trait DeviceOperator: Clone {
     async fn unsubscribe(&self, request: DeviceUnsubscribeRequest);
 }
 
+#[derive(Debug, Clone)]
+pub struct DeviceResponseSubscription {
+    pub sub_id: Option<String>,
+    pub handlers: Vec<mpsc::Sender<DeviceResponseMessage>>,
+}
+
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum DeviceChannelRequest {
     Call(DeviceCallRequest),
     Subscribe(DeviceSubscribeRequest),
     Unsubscribe(DeviceUnsubscribeRequest),
 }
 
+impl DeviceChannelRequest {
+    pub fn get_callsign_method(&self) -> (String, String) {
+        match self {
+            DeviceChannelRequest::Call(c) => {
+                let mut collection: Vec<&str> = c.method.split('.').collect();
+                let method = collection.pop().unwrap_or_default();
+                let callsign = collection.join(".");
+                (callsign, method.into())
+            }
+            DeviceChannelRequest::Subscribe(s) => (s.module.clone(), s.event_name.clone()),
+            DeviceChannelRequest::Unsubscribe(u) => (u.module.clone(), u.event_name.clone()),
+        }
+    }
+
+    pub fn is_subscription(&self) -> bool {
```

Comment:
I don't think `is_subscription()` and `is_unsubscribe()` are used anywhere, if not can we remove?